### PR TITLE
Updating the Kafka pack to the 2.x format. Initial commit to exchange

### PR DESCRIPTION
**Pull request by lmarty**
Updated the Kafka pack to the 2.x pack format. 

- Changed repo structure to contain all artifacts in the top level
- Created config schema in 2.x style
- Created config example file

### CODE REVIEW

**Comment by LindsayHill** [comment]
Nice work! Thanks @lmarty

File: kafka/pack.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+name: kafka
+description: Integration pack for Apache Kafka message broker
+keywords:
+  - kafka
+  - queuing
+  - messaging
+  - message broker
+version: 0.1.0
```
Comment (LindsayHill) [inline kafka/pack.yaml]: Can you bump this version up to 0.2.0? So it doesn't get confused with https://github.com/StackStorm/st2-kafka

You could also add yourself as a contributor? (Or not, up to you)

**Comment by lmarty** [comment]
@LindsayHill Changes added

**Comment by lmarty** [comment]
SYN?

**Comment by rmoshea** [comment]
Awesome work @lmarty

**Comment by LindsayHill** [comment]
Thanks for the good work @lmarty, I will be setting up a new stackstorm-kafka repo and transferring this today

**Comment by armab** [comment]
IMO to benefit from the GitHub repo change (redirects) it might be better to transfer it to exchange org: 
https://github.com/stackstorm/st2-kafka -> https://github.com/stackstorm-exchange/stackstorm-kafka and cherry pick fixes made by @lmarty.

This way users trying to install pack from the old repo will be redirected to new exchange one.

This will also close https://github.com/StackStorm/st2-kafka/issues/9

**Comment by LindsayHill** [comment]
@lmarty I have transferred the pack to github.com/StackStorm-Exchange/stackstorm-kafka. I fixed a couple of small linting errors. 

Once the index gets rebuilt (should be shortly), you'll be able to install it with `st2 pack install`. Can you check that it all looks good?

**Pull request closed without merging**